Chat screen markup

robot269006 opened the pull request:
# WHAT
Chat screen markup

# WHY
1. Implicate haml, BEM, sass
2. In order to get the image of the front page

robot269006 committed:
Import stylesheet reset

robot269006 committed:
Create _reset.scss

1. Created new file named “_reset.scss”
2. Pasted the resetting code from TECH::CAMP website

robot269006 committed:
Setting minimum routing

1. Added “root ‘top#index’” to routes.rb
2. Created top_controller
3. Created index.html.haml file within the top folder

robot269006 committed:
Haml markup

1. Added def index to top_controller
2. Index view file markup

robot269006 committed:
Initial settings before styling

1. Set imports on application.css
2. Created setting files (_file), and stylings files (files with no
underscores)
3. Set variables
4. Set mixins

robot269006 committed:
font-awesome install

robot269006 committed:
Fix import settings

There was a file name error:
Before -> “mixin.scss”
After -> “mixins.scss”

robot269006 committed:
Buttons markup

1. markup of class “editbutton”
2. markup of class “sendbutton”

*Buttons are styled first instead of body, in order to test out the
mixin and variable settings

robot269006 committed:
Fix font-awesome

1. Added @import “font-awesome”
2. font-awesome description changed

robot269006 committed:
Icons markup

1. Icon classes marked up
2. all icons are added with “float” property

robot269006 committed:
Body markup

1. Body markup
2. Delete clearfix mixin as not needed
3. Added property revision and addition to buttons and icons
4. Added some revisions on view file
5. Deleted font-awesome directory since CDN is applied instead.

robot269006 committed:
Width adjustment

kenchan0503 commented on app/assets/stylesheets/_mixins.scss:
```diff
@@ -0,0 +1,11 @@
+//  display: both;
+//}
+
+//settings of clearfix
```
改行追加しましょう

robot269006 replied:
レビューありがとうございます、対応しました

kenchan0503 commented on app/assets/stylesheets/_variables.scss:
```diff
@@ -0,0 +1,2 @@
+$blue-bg: #38AEF0;
+$white-bg: #FFFFFF;
```
改行追加しましょう

kenchan0503 replied:
#FFFFFF は #FFFと省略することが可能です

robot269006 replied:
レビューありがとうございます、対応しました

kenchan0503 commented on test/controllers/top_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
testファイルは生成されないように設定し、こちらのファイルも削除しましょう

robot269006 replied:
ファイルは削除しました。testファイルを生成されないようにするにはどうすればいいですか？一通り検索したのですがファイルを自動生成します、というソースしか見つからないです。

KoichiOzaki replied:
この辺りの記述を参考にしてみてください。
http://qiita.com/_keisuke/items/9b80308aa37afd6899a7
http://ryouzis.hatenablog.com/entry/2015/08/18/185851

robot269006 replied:
ありがとうございます！参考にさせていただきました。

robot269006 committed:
Minor revision

1. Added blank row at the end of the codes for mixins.scss and
variable.scss
2. Deleted top_controller_test

robot269006 committed:
Merge branch 'chat-screen-markup' into user-management-implementation

KoichiOzaki commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,16 @@
+//Styling
+@import "button.scss";
+@import "icon.scss";
+@import "body.scss";
```
ファイルの末尾改行しましょう。

robot269006 replied:
修正しました

KoichiOzaki commented on app/assets/stylesheets/body.scss:
```diff
@@ -0,0 +1,105 @@
+      background-color: #AAAAAA;
+      padding: 20px 40px;
+      .write {
+        width: calc(100vw - 300px - 80px);
```
300px,80pxなどの固定値は変数化しておくと良いです。
今の状態ではnav barの長さを変更すると全てのcalcを変更しないといけないと思うので。

robot269006 replied:
修正しました。が、なぜか横幅が短くなってしまいます。
.groupという右半分のブロックに対してwidth: calc(100vw - $nav-width);とスタイル指定していて、変数 $nav-widthは_variables.scssで $nav-width: 300px;と指定しています。

kenchan0503 replied:
変数が読み込まれていない可能性がありますね。
きちんと変数が読み込まれているか確認しましょう！

robot269006 committed:
Merge branch 'user-management-implementation' into chat-screen-markup

robot269006 committed:
Added "break" at the end of files

robot269006 committed:
Configured not to auto-generate test files

robot269006 committed:
Replace px points with variables

robot269006 committed:
Width variable set with #{$nav-width}
